Exception on /gm/gmtools/announcement [GET]
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/site-packages/flask/app.py", line 1982, in wsgi_app
    response = self.full_dispatch_request()
  File "/usr/local/lib/python2.7/site-packages/flask/app.py", line 1614, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/usr/local/lib/python2.7/site-packages/flask/app.py", line 1517, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/usr/local/lib/python2.7/site-packages/flask/app.py", line 1612, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python2.7/site-packages/flask/app.py", line 1598, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/usr/local/lib/python2.7/site-packages/flask_login/utils.py", line 228, in decorated_view
    return func(*args, **kwargs)
  File "/Users/zouyang/PycharmProjects/GMTools_Flask/route.py", line 131, in gm_tools_announcement
    return render_template('gm_tools_announcement.html', announcement_form=None, cos_path=None)
  File "/usr/local/lib/python2.7/site-packages/flask/templating.py", line 134, in render_template
    context, ctx.app)
  File "/usr/local/lib/python2.7/site-packages/flask/templating.py", line 116, in _render
    rv = template.render(context)
  File "/usr/local/lib/python2.7/site-packages/jinja2/environment.py", line 1008, in render
    return self.environment.handle_exception(exc_info, True)
  File "/usr/local/lib/python2.7/site-packages/jinja2/environment.py", line 780, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "/Users/zouyang/PycharmProjects/GMTools_Flask/templates/gm_tools_announcement.html", line 3, in top-level template code
    {% extends 'gm_tools_layout.html' %}
  File "/Users/zouyang/PycharmProjects/GMTools_Flask/templates/gm_tools_layout.html", line 23, in top-level template code
    {% extends 'side_nav_bar_layout.html' %}
  File "/Users/zouyang/PycharmProjects/GMTools_Flask/templates/side_nav_bar_layout.html", line 1, in top-level template code
    {% extends 'layout.html' %}
  File "/Users/zouyang/PycharmProjects/GMTools_Flask/templates/layout.html", line 8, in top-level template code
    {% set authority_list = [('none','none'),('gm_1','gm_1'),('gm_2','gm_2'),('gm_3','gm_3')] %}
  File "/Users/zouyang/PycharmProjects/GMTools_Flask/templates/base.html", line 1, in top-level template code
    {% block doc -%}
  File "/Users/zouyang/PycharmProjects/GMTools_Flask/templates/base.html", line 4, in block "doc"
    {%- block html %}
  File "/Users/zouyang/PycharmProjects/GMTools_Flask/templates/base.html", line 21, in block "html"
    {% block body -%}
  File "/Users/zouyang/PycharmProjects/GMTools_Flask/templates/base.html", line 24, in block "body"
    {% block content -%}
  File "/Users/zouyang/PycharmProjects/GMTools_Flask/templates/layout.html", line 54, in block "content"
    {% block page_content %}
  File "/Users/zouyang/PycharmProjects/GMTools_Flask/templates/side_nav_bar_layout.html", line 7, in block "page_content"
    {% block side_nav_bar %}
  File "/Users/zouyang/PycharmProjects/GMTools_Flask/templates/side_nav_bar_layout.html", line 31, in block "side_nav_bar"
    {% block main_content %}
  File "/Users/zouyang/PycharmProjects/GMTools_Flask/templates/gm_tools_announcement.html", line 30, in block "main_content"
    {% for key in cos_path %}
TypeError: 'NoneType' object is not iterable
Exception on /gm/gmtools/announcement [POST]
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/site-packages/flask/app.py", line 1982, in wsgi_app
    response = self.full_dispatch_request()
  File "/usr/local/lib/python2.7/site-packages/flask/app.py", line 1614, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/usr/local/lib/python2.7/site-packages/flask/app.py", line 1517, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/usr/local/lib/python2.7/site-packages/flask/app.py", line 1612, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python2.7/site-packages/flask/app.py", line 1598, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/usr/local/lib/python2.7/site-packages/flask_login/utils.py", line 228, in decorated_view
    return func(*args, **kwargs)
  File "/Users/zouyang/PycharmProjects/GMTools_Flask/route.py", line 140, in gm_tools_announcement
    f.write(content)
TypeError: expected a string or other character buffer object
